polaris-react/src/components/Button/Button.scss

```
@@ -893,6 +889,11 @@
   &.iconOnly {
     margin: calc(-1 * var(--p-space-2));
 
+    #{$se23} & {
+      // stylelint-disable-next-line -- custom px to align icons with line height
+      margin: calc(-1 * 3px) calc(-1 * var(--p-space-1_5-experimental));
```

Can you add a comment on where the 3px value is coming from?

```suggestion
      margin: -3px calc(-1 * var(--p-space-1_5-experimental));
```

---

I updated it with a different approach that doesn't use this custom hacky 3px! Basically I was doing this to align the icon with the line height, but actually zeroing out the margins and padding then making the `min-height` = `line-height-2` did the trick :D This is also great so that the wrapping divs can calculate the correct height of the button since we aren't using negative margins on it anymore